src/interpreter/initialization.rs

```diff
@@ -13,11 +13,10 @@ use std::io;
 
 impl<S> Interpreter<S> {
     /// Initialize the VM with a given transaction
-    pub fn init(&mut self, predicate: bool, block_height: u32, tx: CheckedTransaction) -> Result<(), InterpreterError> {
+    pub fn init(&mut self, block_height: u32, tx: CheckedTransaction) -> Result<(), InterpreterError> {
```

Comment:
Block height seems to be something that belongs to the context, and not to VM initialization. Should we tackle that in this PR?

Reply:
It seems like we shouldn't be setting block_height on the VM at all, especially now that we moved the transaction validation logic out of VM initialization I don't think it would be used. The block height opcode should be able to just fetch it from interpreter storage as needed.

Reply:
https://github.com/FuelLabs/fuel-vm/pull/174/commits/3716e92f8a45823387c8fc380bc7fdcdbd3eb287

Reply:
The previous implementation of the context isn't good. Also, the block height belongs to the context and should be fetched from the storage a single time (on script init since block height isn't valid for predicates).

We shouldn't fetch the block height from the storage during runtime because this is a fallible implementation from the storage:

https://github.com/FuelLabs/fuel-vm/pull/174/commits/2f781e30e190c82ce1c0ca194992435d22f6e363